src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,111 @@
+from src.autotester.server.utils.file_management import *
+import os.path
+import re
+import tempfile
+import shutil
+from src.autotester.config import config
+FILES_DIRNAME = config["_workspace_contents", "_files_dir"]
+
+def test_clean_dir_name():
+    a = "markus/address"
+    b = a.replace("/", "_")
+    assert clean_dir_name(a) == b
+
+
+def test_random_tmpfile_name():
+    tmp_file = random_tmpfile_name()
+    dir_name = os.path.dirname(tmp_file)
+    base_name = os.path.basename(tmp_file)
+    assert os.path.exists(dir_name) and re.match(
+        r"[+\-]" "0" "[xX]" "0|" "([1-9A-Fa-f][1-9A-Fa-f]*)", base_name
```

Comment:
can you remind me where this regular expression is from? (and why it's being used in this test?)

Reply:
random_tmpfile_name() returns a temporary directory as dirname and hexadecimal UUID as a filename. The regular expression checks the UUID (filename)

Reply:
ok that's fine but I think the more important thing to test here is that the filename is random (ie. its different every time), not that the filename corresponds to a specific format.